src/utils/transactions/TransactionResultUtil.ts

```diff
@@ -20,14 +20,96 @@ export enum EcommerceFinalStatusCodeEnum {
   NOTIFIED_KO,
 }
 
+export enum NexiResultCodeEnum {
+  SUCCESS = "0",
+  INCORRECT_PARAMS = "1",
+  NOT_FOUND = "2",
+  INCORRECT_MAC = "3",
+  MAC_NOT_PRESENT = "4",
+  TIMEOUT = "5",
+  INVALID_APIKEY = "7",
+  INVALID_CONTRACT = "8",
+  DUPLICATE_TRANSACTION = "9",
+  INVALID_GROUP = "12",
+  TRANSACTION_NOT_FOUND = "13",
+  EXPIRED_CARD = "14",
+  CARD_BRAND_NOT_PERMITTED = "15",
+  INVALID_STATUS = "16",
+  EXCESSIVE_AMOUNT = "17",
+  RETRY_EXHAUSTED = "18",
+  REFUSED_PAYMENT = "19",
+  CANCELED_3DS_AUTH = "20",
+  FAILED_3DS_AUTH = "21",
+  INVALID_CARD = "22",
+  INVALID_MAC_ALIAS = "50",
+  KO_RETRIABLE = "96",
+  GENERIC_ERROR = "97",
+  UNAVAILABLE_METHOD = "98",
+  FORBIDDEN_OPERATION = "99",
+  INTERNAL_ERROR = "100",
+}
+
+export enum VposResultCodeEnum {
+  SUCCESS = "00",
+  ORDER_OR_REQREFNUM_NOT_FOUND = "01",
+  REQREFNUM_INVALID = "02",
+  INCORRECT_FORMAT = "03",
+  INCORRECT_MAC_OR_TIMESTAMP = "04",
+  INCORRECT_DATE = "05",
+  UNKNOWN_ERROR = "06",
+  TRANSACTION_ID_NOT_FOUND = "07",
+  OPERATOR_NOT_FOUND = "08",
+  TRANSACTION_ID_NOT_CONSISTENT = "09",
+  EXCEEDING_AMOUNT = "10",
+  INCORRECT_STATUS = "11",
+  CIRCUIT_DISABLED = "12",
+  DUPLICATED_ORDER = "13",
+  UNSUPPORTED_CURRENCY = "16",
+  UNSUPPORTED_EXPONENT = "17",
+  REDIRECTION_3DS1 = "20",
+  TIMEOUT = "21",
+  METHOD_REQUESTED = "25",
+  CHALLENGE_REQUESTED = "26",
+  PAYMENT_INSTRUMENT_NOT_ACCEPTED = "35",
+  MISSING_CVV2 = "37",
+  INVALID_PAN = "38",
+  XML_EMPTY = "40",
+  XML_NOT_PARSABLE = "41",
+  INSTALLMENTS_NOT_AVAILABLE = "50",
+  INSTALLMENT_NUMBER_OUT_OF_BOUNDS = "51",
+  APPLICATION_ERROR = "98",
+  TRANSACTION_FAILED = "99",
+}
+
 export const getViewOutcomeFromEcommerceResultCode = (
-  ecommerceStatus?: TransactionStatusEnum
+  transactionStatus?: TransactionStatusEnum,
+  authorizationResult?: string,
+  gateway?: string,
+  errorCode?: string
 ): ViewOutcomeEnum => {
-  switch (ecommerceStatus) {
+  switch (transactionStatus) {
     case TransactionStatusEnum.NOTIFIED_OK:
       return ViewOutcomeEnum.SUCCESS;
+    case TransactionStatusEnum.NOTIFICATION_REQUESTED:
+    case TransactionStatusEnum.NOTIFICATION_ERROR:
+      return authorizationResult === "OK"
```

Comment:
Shold this check be performed against nodo sendPaymentResult outcome? transaction can go to `NOTIFICATION_REQUESTED` and `NOTIFICATION_ERROR` only if it was authorized, so authorizationResult will be always OK here

Reply:
Indeed this is the sendPaymentResult outcome. You raised a possible point of confusion,but since the sendPaymentResultOutcome is named AuthorizationResultDto we named it authorizationResult in NewTransactionResponse in transaction-service